articles/most-useful-nodejs-packages/index.md

```diff
@@ -0,0 +1,424 @@
+
+Nodemon is simple to use. You just need to replace the word node with `nodemon` on the command-line when executing your `script`, i.e. `nodemon ./server.js localhost 300`
+
+#### Example of Other Monitoring Frameworks Include
+-   [`PM2`](https://www.npmjs.com/package/pm2) - is a Node.js process manager that allows you to keep application alive forever and reload with no downtime.
+-   [`Trace`](https://github.com/RisingStack/trace-nodejs) - is designed for microservices and is a visualized stack trace platform.
+-   [`Forever`](https://www.npmjs.com/package/forever) - is a CLI tool that ensures Node.js scripts runs continuously i.e. forever.
+
+### 12. Restify
+[Restify](http://restify.com/) is a Node.js web service framework, which is optimized for building semantic and RESTful web services (APIs) that are ready for production use at scale.
+
+#### Why You Should Use Restify
+-   Production-ready - Optimized for introspection and performance.
+-   Debuggable - Trace problems back to the origin. It is built from a post-mortem debugging perspective.
+-   Semantically correct - It has references littered all-over GitHub and the codebase.
+
+#### Restify Examples
+
+##### i. Server
+
+```js
+var restify = require('restify');
+const server = restify.createServer({
+    name: 'myapp',
+    version: '1.0.0'
+});
+server.use(restify.plugins.acceptParser(server.acceptable));
+server.use(restify.plugins.queryParser());
+server.use(restify.plugins.bodyParser());
+server.get('/echo/:name', function (req, res, next) {
+    res.send(req.params);
+    return next();
+});
+server.listen(8080, function () {
+    console.log('%s listening at %s', server.name, server.url);
+});
+```
+[Code source](https://github.com/restify/node-restify).
+
+##### ii. Client
+
+```js
+var assert = require('assert');
+var clients = require('restify-clients');
+var client = clients.createJsonClient({
+    url: 'http://localhost:8080',
+    version: '~1.0'
+});
+client.get('/echo/mark', function (err, req, res, obj) {
+    assert.ifError(err);
+    console.log('Server returned: %j', obj);
+});
+```
+[Code source](https://github.com/restify/node-restify).
+
+##### iii. Routing
+
+```js
+var Router = require('restify-router').Router;
+var routerInstance = new  Router();
+var restify = require('restify');
+ 
+function respond(req, res, next) {
+  res.send('hello ' + req.params.name);
+  next();
+}
+// add a route like you would on a restify server instance
+routerInstance.get('/hello/:name', respond);
+var server = restify.createServer();
+// add all routes registered in the router to this server instance
+routerInstance.applyRoutes(server);
+server.listen(8080, function() {
+  console.log('%s listening at %s', server.name, server.url);
+});
+```
```

Comment:
If users are supposed to run these three code blocks then you should provide more instructions. I couldn't run it.

Reply:
@louisefindlay23 I opted to remove the code block and explained the subtopics instead.